src/mocks/data.ts

```diff
@@ -305,13 +305,13 @@ export const registrationUserData = {
   verificationInfo: "www.website.com",
 };
 
-export const mockUser = {
+export const mockUser: RegisteredUserDTO = {
   givenName: "Test",
   familyName: "Admin",
   email: "test-admin@test.com",
-  dateOfBirth: 777777777777,
+  dateOfBirth: 777777777777 as unknown as Date,
```

Comment:
Doesn't this mean that `RegisteredUserDTO` has the wrong type for `dateOfBirth`, `registrationDate`, etc? They're clearly numbers, so doesn't that mean that they should be changed from Date to number in the `RegisteredUserDTO` type itself, rather than here?

Reply:
They're Dates in the DB and API (and all the types appear to expect Dates for date fields in the React App). But when the response comes back from API it looks like this 

![Screenshot 2023-10-30 at 09 03 20](https://github.com/isaaccomputerscience/isaac-react-app/assets/114683476/78a7b12b-74cc-46a0-97f8-bffce78e151f)

Probably a ticket in its own right if I have to go through and change everywhere that Date was chosen but ought to be Number 😀

Reply:
Thanks for clarifying. If that's how the response looks like, then in your React code the type should be `number` (or, even better, `EpochTimeStamp`, which is an alias for `number`, but it more clearly states the purpose of said number). As you said, this should be a ticket on its own right, so you don't need to go and fix it everywhere, but I would suggest fixing it in the places were you've added `as unknown as Date` in this PR. Otherwise, you'll be contributing to the problem.

Reply:
Not clear on how to correct this without some fairly widespread changes - to get rid of as unknown as Date here I would need to modify the `RegisteredUserDTO` for `dateOfBirth`, `registrationDate`, `registeredContextsLastConfirmed`, `lastUpdated`, `lastSeen`, and given that this type is used across the app, I suspect a lot of components will require changes also.  I did a quick test run and it made all 40 test suites fail, and the app gets a bunch of compile errors 🤨

Reply:
If this is hard to fix, then this is a worrying issue, because it could mean that we may be using Dates and numbers as the type of the same attribute in different places, and that is just really messy. Could you at least fix the easy ones that have just a few usages, like `registrationDate`, please? But only for the ones that you've added `as unknown as Date` in this PR. And please, double check that the actual value that we get from the Java API is an integer. If we never convert those timestamps to actual JS Dates, then the type of those values should definitely be `EpochTimeStamp`. But we need to confirm this.

And next sprint we have to fix all the misuses of `Date`, this is very important. We need to do this next sprint, because I don't like the idea of adding dangerous workarounds in purpose like `as unknown as Date`, increasing our already too big tech debt.

Reply:
I stuck a console.log in for the fetch of `currentUser` and React sees the `registrationDate` as a `number` when it is received. `registrationDate` is only used across the app in a handful of places so changing that data type to `EpochTimeStamp` appears to be fine.

Reply:
Corrected a few of them now, though not `registrationDate` and `registeredContextsLastConfirmed` as they will need other changes made. There's widespread usage of Date in the types where I suspect they're probably all incorrect, but I'll create a ticket.

Reply:
https://github.com/isaaccomputerscience/isaac-cs-issues/issues/196